kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -406,6 +431,11 @@ namespace Kratos
 
         // Correct the distance values orientation
         CorrectDistanceOrientation(r_geometry, rIntersectedObjects, r_elemental_distances);
+
+        // Replace zero values of the distance with an epislon
+        if (mOptions.IsNot(CalculateDiscontinuousDistanceToSkinProcessFlags::USE_ZERO_DISTANCE_VALUES)) {
+            ReplaceZeroDistances(r_elemental_distances);
```

Comment:
A flag is used here to replace or not the zero distance values with an epsilon. By default, `USE_ZERO_DISTANCE_VALUES = false`